pkg/cmd/roachtest/test_runner.go

```diff
@@ -1723,6 +1736,79 @@ func (r *testRunner) teardownTest(
 	return nil
 }
 
+// inspectArtifacts inspects node logs and attempts to write helpful triage
+// information to the test log and testRunner.githubMessage
+// This method is best effort and should not fail a test.
+func (r *testRunner) inspectArtifacts(
+	ctx context.Context, t *testImpl, testLogger *logger.Logger, timedOut bool,
+) error {
+
+	if timedOut {
+		testLogger.Printf("Test timed out. Do not attempt to inspect artifacts.")
+		return nil
+	}
+
+	var errMonitor registry.FatalMonitorError
+	if !failuresMatchingError(t.failures(), &errMonitor) {
+		return nil
+	}
+
+	testLogger.Printf("Test has failed due to fatal error observed by the monitor. " +
+		"Attempting to gather node fatal level logs for triage.")
+	out, err := gatherFatalNodeLogs(t, testLogger)
+	if err != nil {
+		return err
+	}
+
+	msg := fmt.Sprintf("CockroachDB contains Fatal level logs. Up to the first 10 "+
+		"will be shown here. See node logs in artifacts for more details.\n%s", out)
+	testLogger.PrintfCtx(ctx, msg)
+	t.appendGithubMessage(msg)
+
+	return nil
+}
+
+// gatherFatalNodeLogs attempts to gather fatal level node logs to help with
+// triage
+func gatherFatalNodeLogs(t *testImpl, testLogger *logger.Logger) (string, error) {
+	logPattern := `^F[0-9]{6}`
+	filePattern := "logs/*unredacted/cockroach*.log"
```

Comment:
This will match other unrelated things like `cockroach.stdout.log` and `cockroach-health.log`. It seems okay to expand the search radius, but probably worth adding a comment to clarify the glob expression.